snap/hooks/configure

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+FEATURE_GATES="$(snapctl get feature-gates)"
```

Comment:
There is a semi established [pattern for snap config](https://documentation.ubuntu.com/snapcraft/latest/how-to/crafting/add-a-snap-configuration/), using `get_`, `set_` and `handle_` ([example](https://github.com/canonical/grafana-agent-snap/blob/main/snap/hooks/configure)). Let's follow it here as well?

Reply:
nothing else about this snap follows that convention though (including the other existing config-files parameter), so not only that's a lot of work, but also it would mean breaking in-project consistency which IMO beats ecosystem consistency. 
I say refactoring the snap to do this is out of scope for this PR.

Reply:
adding https://github.com/canonical/opentelemetry-collector-snap/issues/36 to track